syntax = "proto3";

package crowdsourcing;

message TextImage {
    optional string text = 1;
    optional string image_path = 2;
}

enum AnswerType {
  TEXT = 0;
  IMAGE = 1;
}

message TaskPublish {
  uint32 task_id = 1;
  uint32 publisher_id =2;
  uint32 type = 3;
  string name = 4;
  string description = 5;
  uint32 publish_time = 6;
  uint32 expire_time = 7;
  uint32 duplicate_num = 8;

  message Question{
    repeated TextImage text_images = 1;
    AnswerType expected_answer_type = 2;
  }
  repeated Question questions = 9;
}

message TaskFinish {
  uint32 task_id = 1;
  uint32 worker_id = 2;
  uint32 lat = 4;
  uint32 long = 5;
  uint32 update_time = 6;
  message Answer {
    repeated TextImage text_images = 1;
    AnswerType answer_type = 2;
  }
  repeated Answer answers = 9;
}

message Worker {
  uint32 id = 1;
  uint32 type = 2;
  string name = 3;
}

message Publisher {
  uint32 id = 1;
  uint32 type = 2;
  string name = 3;
}